src/libmongoc/src/mongoc/mongoc-http.c

```diff
@@ -127,8 +126,8 @@ _mongoc_http_send(const mongoc_http_request_t *req,
    }
 
    stream = mongoc_client_connect_tcp(
-      // +1 to prevent passing zero as a timeout
-      _mongoc_http_msec_remaining(timer) + 1,
+      // Prevent passing zero as a timeout
+      mlib_time_cmp(timer.expires_at, >, (mlib_time_point){0}) ? _mongoc_http_msec_remaining(timer) : 1,
```

Comment:
```suggestion
      // Prevent passing zero as a timeout
      BSON_MAX(_mongoc_http_msec_remaining(timer), 1),
```

Sorry, my prior suggestion was incorrect: this mixes up the `expires_at` time point with a duration.